### PR TITLE
Serve PTR records for unambiguous tasks

### PR DESCRIPTION
**Pull request by christian-marie**
Hi, we (a team at Commonwealth Bank) are trying to roll out a HDFS cluster on top of Mesos. This seems to require reverse DNS to be properly configured for tasks, and this seemed the most convenient place to add PTR records.

This patch is currently only unit tested, I'll be integration testing tomorrow.

This may be of interest to #286 and #67 (closed).

### CODE REVIEW

**Comment by jdef** [comment]
ok to test

**Comment by christian-marie** [comment]
Update: Handlers needed to be set up for these reverse records (obviously).

Tested to work in practice now (very ad-hoc testing, redacted):

```
$ dig aaa.bb.cc.dd.in-addr.arpa. -tptr @localhost

; <<>> DiG 9.10.3-P4-Ubuntu <<>> aaa.bb.cc.dd.in-addr.arpa. -tptr @localhost
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 29836
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;aaa.bb.cc.dd.in-addr.arpa.	IN	PTR

;; ANSWER SECTION:
aaa.bb.cc.dd.in-addr.arpa. 15	IN	PTR	task-name.domain.

;; Query time: 0 msec
;; SERVER: ::1#53(::1)
;; WHEN: Mon Oct 17 05:08:54 UTC 2016
;; MSG SIZE  rcvd: 93
```

**Comment by christian-marie** [comment]
@jdef I'm pretty happy with this now, does it seem workable?

**Comment by jdef** [comment]
last time @sargun that I discussed PTR records there was some concern about mesos-dns being responsible for ALL PTR-resolution and that it would break in enterprise environments. We really didn't have a strong plan for delegation of PTR resolution. There's at least one RFC that tries to tackle this. And a quick google search turns up existing documentation for other DNS apps w/ respect to PTR delegation, e.g. http://www.zytrax.com/books/dns/ch9/reverse.html . I suspect that a more robust solution would be needed long term if we intent to fully support PTR resolution in reasonably sized clusters within an enterprise context.

I have no problem with mesos-dns supporting PTR resolver functionality, as implemented, as long as it defaults to "off" (seems to be the case for this PR) and comes with a strong warning in the docs that this is a highly experimental feature and enabling it could break anything but the most simple mesos clusters.

**Comment by christian-marie** [comment]
Ok. I'll add some experimental flags and docs today.

**Comment by christian-marie** [comment]
Rebased, WIP removed.

**Comment by christian-marie** [comment]
P.S. Some docs added.

**Comment by sargun** [comment]
Is there a SOA for the PTR zone?

**Comment by christian-marie** [comment]
Not sure what you mean. If you can modify the test to demonstrate, that would hep.

**Comment by christian-marie** [comment]
Good day sirs

File: records/generator.go [hunk 1]
```diff
@@ -554,6 +561,10 @@ func (rg *RecordGenerator) taskContextRecord(ctx context, task state.Task, f sta
 	rg.insertTaskRR(arec+tail, ctx.taskIP, A, enumTask)
 	rg.insertTaskRR(canonical+tail, ctx.taskIP, A, enumTask)
 
+	// insert PTR records, these are only served when ReverseDnsOn is true,
+	// which sets up a handler for the reveverse DNS zones
+	rg.insertTaskRR(arec+tail, ctx.taskIP, PTR, enumTask)
```
Comment (drewkerrigan) [inline records/generator.go]: @christian-marie Perhaps we should only insert this type when configured to do so? Wouldn't want unusable data taking up space.

Comment (christian-marie) [reply records/generator.go]: Don't really care about performance or this patch anymore.

**Comment by jdef** [comment]
closing this out since the OP is no longer interested in this and we haven't needed it yet for DC/OS. please re-open if this is being closed in error.

**Pull request closed without merging**